pkg/devfile/generator/generators.go

```diff
@@ -91,6 +98,70 @@ func GetContainers(devfileObj parser.DevfileObj, options common.DevfileOptions)
 	return containers, nil
 }
 
+// GetInitContainers gets the init container for every preStart devfile event
+func GetInitContainers(devfileObj parser.DevfileObj) ([]corev1.Container, error) {
+	containers, err := GetContainers(devfileObj, common.DevfileOptions{})
+	if err != nil {
+		return nil, err
+	}
+	preStartEvents := devfileObj.Data.GetEvents().PreStart
+	var initContainers []corev1.Container
+	if len(preStartEvents) > 0 {
+		var eventCommands []string
+		commands, err := devfileObj.Data.GetCommands(common.DevfileOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		commandsMap := getCommandsMap(commands)
+
+		for _, event := range preStartEvents {
+			eventSubCommands := getCommandsFromEvent(commandsMap, strings.ToLower(event))
+			eventCommands = append(eventCommands, eventSubCommands...)
+		}
+
+		for i, commandName := range eventCommands {
+			if command, ok := commandsMap[commandName]; ok {
+				component := common.GetExecComponent(command)
+				commandLine := common.GetExecCommandLine(command)
+				workingDir := common.GetExecWorkingDir(command)
+
+				var cmdArr []string
+				if workingDir != "" {
+					// since we are using /bin/sh -c, the command needs to be within a single double quote instance, for example "cd /tmp && pwd"
+					cmdArr = []string{shellExecutable, "-c", "cd " + workingDir + " && " + commandLine}
+				} else {
+					cmdArr = []string{shellExecutable, "-c", commandLine}
+				}
```

Comment:
we can no longer do this because of `apply`

Reply:
updated